src/Adapter/Builder/GridFSAdapterDefinitionBuilder.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the flysystem-bundle project.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\FlysystemBundle\Adapter\Builder;
+
+use Doctrine\ODM\MongoDB\DocumentManager;
+use League\Flysystem\GridFS\GridFSAdapter;
+use MongoDB\Client;
+use MongoDB\GridFS\Bucket;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ *
+ * @internal
+ */
+class GridFSAdapterDefinitionBuilder extends AbstractAdapterDefinitionBuilder
+{
+    public function getName(): string
+    {
+        return 'gridfs';
+    }
+
+    protected function getRequiredPackages(): array
+    {
+        return [
+            GridFSAdapter::class => 'league/flysystem-gridfs',
+        ];
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->define('bucket')->default(null)->allowedTypes('string', 'null');
+        $resolver->define('prefix')->default('')->allowedTypes('string');
+        $resolver->define('database')->default(null)->allowedTypes('string', 'null');
+        $resolver->define('doctrine_connection')->allowedTypes('string');
+        $resolver->define('mongodb_uri')->allowedTypes('string');
+        $resolver->define('mongodb_uri_options')->default([])->allowedTypes('array');
+        $resolver->define('mongodb_driver_options')->default([])->allowedTypes('array');
+    }
+
+    /**
+     * @param array{bucket:string|null, prefix:string, database:string|null, doctrine_connection?:string, mongodb_uri?:string, mongodb_uri_options:array, mongodb_driver_options:array} $options
+     */
+    protected function configureDefinition(Definition $definition, array $options, ?string $defaultVisibilityForDirectories): void
+    {
+        if (isset($options['doctrine_connection'])) {
```

Comment:
Wouldn't we have to throw an exception if `doctrine_connection` & `mongodb_uri` are both configured?

Reply:
With the 3 examples, this file would become very long and the navigation rather blurred. Technically it's not a "cloud storage provider", but a "database storage", it's not linked to a specific provider since the database can be hosted anywhere and the protocol used is implemented by other services like Azure Cosmos or Amazon DocumentDB (which are clouds).

That's similar to `3-interacting-with-ftp-and-sftp-servers.md`.

Reply:
Validation added.